### PR TITLE
feat(`campaign`): add `CampaignSummary` and `CampaignSummaries` queries

### PR DESCRIPTION
**Pull request by lubtd**
Used by the client. The queries allow to retrieve a campaign with some information associated with it:

If the campaign has at least one chain associated, retrieve for the most recent chain:
- launchID
- launchTriggered if the launch of the chain is triggered and requests can no longer be sent
- sourceURL
- sourceHash
- requestNb (pending requests for the time being)
- validatorNb

If a reward pool exist and therefore the chain is incentivized
- Rewards: coins to be distributed
- RewardsDistributed: is the reward pool closed

The PR dependency of `reward` module into `campaign` which is not ideal but this is the requirement we have if we want a more optimized query for the front-end

### USAGE
```
spnd q campaign show-campaign-summary <campaign-id>
spnd q campaign list-campaign-summary
```

### Example of output
```
campaignSummaries:
- campaign:
    allocatedShares: []
    campaignID: "1"
    campaignName: orbit
    coordinatorID: "1"
    dynamicShares: false
    mainnetID: "0"
    mainnetInitialized: false
    metadata: null
    totalShares: []
    totalSupply:
    - amount: "1000000"
      denom: orbit
  hasMostRecentChain: true
  incentivized: true
  mostRecentChain:
    launchID: "3"
    launchTriggered: true
    requestNb: "2"
    sourceHash: "0xbbb"
    sourceURL: orbito.com
    validatorNb: "1"
  rewards:
  - amount: "10"
    denom: uspn
  rewardsDistributed: false
- campaign:
    allocatedShares: []
    campaignID: "2"
    campaignName: orbit
    coordinatorID: "1"
    dynamicShares: false
    mainnetID: "0"
    mainnetInitialized: false
    metadata: null
    totalShares: []
    totalSupply:
    - amount: "1000000"
      denom: orbit
  hasMostRecentChain: false
  incentivized: false
  mostRecentChain:
    launchID: "0"
    launchTriggered: false
    requestNb: "0"
    sourceHash: ""
    sourceURL: ""
    validatorNb: "0"
  rewards: []
  rewardsDistributed: false
pagination:
  next_key: null
  total: "2"
```

### CODE REVIEW

**Comment by lubtd** [comment]
@JoseRFelix 

For what we call "Share Allocation" for a chain, this is part of the rewards that you have in `rewards` field. There is no distinction on-chain between regular tokens "Incentives" and vouchers

To parse the vouchers: a voucher is a token with a denom of the following format: `v/{campaignID}/{voucherName}`
The percentage of share allocation for the chain for the specific voucher is:
`amountOfVoucherInRewards / totalNumberOfShare`
`totalNumberOfShare` is currently set to `100000` but eventually we can create a query to retrieve this value?

File: x/campaign/client/cli/query_campaign_summary.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/spn/x/campaign/types"
+)
+
+func CmdListCampaignSummary() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list-campaign-summary",
+		Short: "List information summarizing campaigns",
+		Args:  cobra.ExactArgs(0),
```
Comment (Pantani) [inline x/campaign/client/cli/query_campaign_summary.go:16]:
```suggestion
		Args:  cobra.NoArgs,
```

File: x/campaign/keeper/grpc_campaign_summary.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+// GetCampaignSummary returns the campaign with summary attached to it like most recent chain and rewards attached to it
+// TODO: add tests https://github.com/tendermint/spn/issues/650
+func (k Keeper) GetCampaignSummary(ctx sdk.Context, campaign types.Campaign) (cs types.CampaignSummary, err error) {
+	cs.Campaign = campaign
+
+	campaignChains, found := k.GetCampaignChains(ctx, campaign.CampaignID)
+	if !found {
+		return cs, fmt.Errorf("chain list not found for existing campaign %d", campaign.CampaignID)
```
[the remark this answers is not in the document]
Comment (lubtd) [reply x/campaign/keeper/grpc_campaign_summary.go:78]: Yes good point. Mind that in this case, this event should never occur, like a critical error but we don't need critical error for queries

File: proto/campaign/query.proto [hunk 3]
```diff
@@ -51,12 +52,22 @@ service Query {
...
   }
 
+  // Queries a campaign summary
+  rpc CampaignSummary(QueryCampaignSummaryRequest) returns (QueryCampaignSummaryResponse) {
+    option (google.api.http).get = "/tendermint/spn/campaign/campaign_summary/{campaignID}";
+  }
+
+  // Queries a list of campaign summaries
+  rpc CampaignSummaries(QueryCampaignSummariesRequest) returns (QueryCampaignSummariesResponse) {
+    option (google.api.http).get = "/tendermint/spn/campaign/campaign_summary";
```
Comment (giunatale) [inline proto/campaign/query.proto:62]: might be misleading as an endpoint (due to the one above). Would change it to `/tendermint/spn/campaign/campaigns_summary`

Comment (giunatale) [inline proto/campaign/query.proto:62]:
```suggestion
    option (google.api.http).get = "/tendermint/spn/campaign/campaigns_summary";
```

File: x/campaign/client/cli/query_campaign_summary.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/spn/x/campaign/types"
+)
+
+func CmdListCampaignSummary() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list-campaign-summary",
+		Short: "List information summarizing campaigns",
```
Comment (giunatale) [inline x/campaign/client/cli/query_campaign_summary.go:15]:
```suggestion
		Short: "List information summarizing all campaigns",
```

**Review by giunatale — CHANGES_REQUESTED** [review]

**Review by Pantani — DISMISSED** [review]

**Comment by lubtd** [comment]
UPDATE: added the rewards from previous chains in `previousRewards`

**Review by giunatale — APPROVED** [review]

**Review by aljo242 — APPROVED** [review]

**Review by Pantani — DISMISSED** [review]

**Pull request merged**